module/config/config.go

```diff
@@ -2,23 +2,29 @@ package config
 
 import (
 	"context"
-	"fmt"
+
 	"github.com/datachainlab/ethereum-ibc-relay-chain/pkg/relay/ethereum"
-	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover"
-	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover/l1"
-	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover/l2"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/hyperledger-labs/yui-relayer/core"
+	"github.com/hyperledger-labs/yui-relayer/coreutil"
 	"github.com/hyperledger-labs/yui-relayer/log"
+
+	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover"
+	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover/l1"
+	"github.com/datachainlab/optimism-ibc-relay-prover/module/prover/l2"
 )
 
 var _ core.ProverConfig = (*ProverConfig)(nil)
 
 func (c *ProverConfig) Build(chain core.Chain) (core.Prover, error) {
-	l2Chain, ok := chain.(*ethereum.Chain)
-	if !ok {
-		return nil, fmt.Errorf("chain type must be %T, not %T", &ethereum.Chain{}, chain)
+	l2Chain, err := coreutil.UnwrapChain[*ethereum.Chain](chain)
+	if err != nil {
+		return nil, err
 	}
+	if err := c.Validate(); err != nil {
+		return nil, err
+	}
+
```

Comment:
Why do we need to call `Validate()` here?

Reply:
Since validate is called before build, this call is unnecessary.

Reply:
@siburu 
In that case, the validate in `ethereum-ibc-relay-prover` may not be necessary either.
https://github.com/datachainlab/ethereum-ibc-relay-prover/blob/main/relay/config.go#L69-L75